### ISSUE TITLE
Textures are wrong on Huawei

### ISSUE THREAD

**Issue by Remilia-Scarlet** [baldurk/renderdoc#1799]
## Description
I used Renderdoc to debug a visual glitch bug on Huawei mobile phone.
However, I found that the value of some textures are wrong.
For example, I have a 1x1 black dummy texture used in Tonemap pass.
The value of the pixel captured from Xiaomi(snapdragon GPU) is (0,0,0,0), Ok it's right.
But the value of the pixel captured from Huawei(Mali GPU) is(0.3…,0.01…,0.2…,0.6…), it's absolutly wrong.
![image](https://user-images.githubusercontent.com/12179044/77815356-47ac6400-70f5-11ea-8865-a5dbf14f8186.png)
But the final image displayed on the phone screen is correct, so I guess this wrong value is a renderdoc issue, not the real value in GPU.
Could you help to check this issue? 
Thank you very much.
I made a demo apk to help reproduce this issue, I wish it can help.
You can download it from my Google drive: https://drive.google.com/file/d/1Bl6UI4bcJ5SDTWdm6pvnmXQY17F5Zkt5/view

## Steps to reproduce
1. Download the apk
2. Run Install_Blank424CppSorce2-armv7-es2.bat to install the apk to the mobile phone.
3. Touch the screen with four fingers to trigger the console, input "toggledrawevents". Otherwise there is no maker in capture.
![image](https://user-images.githubusercontent.com/12179044/77816024-77119f80-70fa-11ea-903b-2e5b358e73ac.png)
4. Capture in Renderdoc, find MobileSceneRenderer->PostProcessing->Tonemap. You can see the black dummy texture is not (0,0,0,0).
![image](https://user-images.githubusercontent.com/12179044/77816100-f7d09b80-70fa-11ea-956c-77fb79ef7340.png)

## Environment

* RenderDoc version: 1.7
* Operating System: Android 8.1.0
* Graphics API: OpenGL ES3.1

Thanks again and wish you a good weekend.

**Comment by Remilia-Scarlet** [comment]
It seems to be a common issue on Huawei phones.
I tried at least 3 different Huawei phones and all of them have this issue.

Just now I captured a rdc file from my Huawei Honor 8X. I uploaded to my Google drive too.
https://drive.google.com/open?id=1zqDGtY2J5T2c-9ztp24m78NEobc94xmD
Perhaps it can help you.

In fact, in addition to black dummy texture, there are many other textures which are wrong, but black dummy is the easiest to find.
For eample the color buffer shows there are many strange white mask on the floor and wall.
![image](https://user-images.githubusercontent.com/12179044/77816801-cd81dc80-7100-11ea-978d-0e952694fe8a.png)
But there are no white mask in final phone screen.
![image](https://user-images.githubusercontent.com/12179044/77816928-daeb9680-7101-11ea-91bb-3abbb6e59905.png)
